addons/sale/models/sale_order.py

```diff
@@ -528,6 +533,36 @@ def _compute_amounts(self):
             order.amount_tax = tax_totals['tax_amount_currency']
             order.amount_total = tax_totals['total_amount_currency']
 
+    @api.depends(
+        'order_line.qty_delivered',
+        'order_line.product_uom_qty',
+    )
+    def _compute_is_unfulfilled(self):
+        for order in self:
+            order.is_unfulfilled = (
+                any(
+                    line.qty_delivered < line.product_uom_qty
+                    for line in order.order_line
+                )
+                and order.state == 'sale'
+            )
+
+    def _search_is_unfulfilled(self, operator, value):
+        if operator not in ('=', '!='):
+            return NotImplemented
+
+        operator = 'any' if operator == '=' else 'not any'
```

Comment:
It could have `= False` or `!= True` we should take both operator and value into consideration to decide domain operator and could change name to avoid overwrite 🙈